src/utils/rfqm_health_check.ts

```diff
@@ -48,3 +57,59 @@ export function transformResultToShortResponse(result: HealthCheckResult): RfqmH
             }),
     };
 }
+
+/**
+ * Runs checks on the SQS queue to detect if there are messages piling up or if there are old messages in the
+ * queue which have expired or will expire soon. `nowTime` parameter is provided for testing.
+ */
+export async function checkSqsQueueAsync(producer: Producer, nowTime: Date = new Date()): Promise<HealthCheckIssue[]> {
+    const results: HealthCheckIssue[] = [];
+    const messagesInQueue = await producer.queueSize();
+    if (messagesInQueue === 0) {
+        return results;
+    }
+    if (messagesInQueue > SQS_QUEUE_SIZE_FAILED_THRESHOLD) {
+        results.push({
+            status: HealthCheckStatus.Failed,
+            description: `SQS queue contains ${messagesInQueue} messages (threshold is ${SQS_QUEUE_SIZE_FAILED_THRESHOLD})`,
+        });
+    } else if (messagesInQueue > SQS_QUEUE_SIZE_DEGRADED_THRESHOLD) {
+        results.push({
+            status: HealthCheckStatus.Degraded,
+            description: `SQS queue contains ${messagesInQueue} messages (threshold is ${SQS_QUEUE_SIZE_DEGRADED_THRESHOLD})`,
+        });
+    }
+    const { Messages: messages } = await producer.sqs
+        .receiveMessage({
```

Comment:
Are we sure that this is doing exactly what we are expecting it to do? Is there any documentation around receiving messages with 0 visibility window? Could we use the Postgres database instead to get this kind of information, instead of peeking the queue?